crates/ide-assists/src/utils.rs

```diff
@@ -289,16 +289,22 @@ pub(crate) fn next_prev() -> impl Iterator<Item = Direction> {
 
 pub(crate) fn does_pat_match_variant(pat: &ast::Pat, var: &ast::Pat) -> bool {
     let first_node_text = |pat: &ast::Pat| pat.syntax().first_child().map(|node| node.text());
+    let last_segment_text =
+        |path: &ast::Path| path.segments().last().map(|segment| segment.syntax().text());
+    let normalize_pat = |pat: &ast::Pat| match pat {
```

Comment:
Could you put a `HACK` or some other kind of comment on this explaining why this is being done in source? Otherwise encountering this will be confusing to people not knowing of the PR.